routes/files.test.js

```diff
@@ -19,29 +24,70 @@ describe('files routes', () => {
       return context.inject({ ...defaults, ...options });
     }
 
-    const pageUrl = 'https://en.wikipedia.org/wiki/Wikimedia_Foundation';
+    it('returns a list of all files from the given article', async () => {
+      const articleUrl = 'https://en.wikipedia.org/wiki/Wikimedia_Foundation';
+      const encodedArticleUrl = encodeURIComponent(articleUrl);
+      files.getPageImages.mockResolvedValue(filesMock);
 
-    describe('with a valid encoded url', () => {
-      const encodedPageUrl = encodeURIComponent(pageUrl);
-      const files = [
-        { file: 'File:image.jpg', url: 'https://en.wikipedia.org/wiki/File:image.jpg' },
-      ];
+      const response = await subject({ url: `/files/${encodedArticleUrl}` });
 
-      it('returns a list of files with their name and url', async () => {
-        service.getPageImages.mockResolvedValue(files);
-        const response = await subject({ url: `/files/${encodedPageUrl}` });
+      expect(files.getPageImages).toHaveBeenCalledWith(articleUrl);
+      expect(response.status).toBe(200);
+      expect(response.type).toBe('application/json');
+      expect(response.payload).toMatchSnapshot();
+    });
 
-        expect(response.status).toBe(200);
-        expect(response.type).toBe('application/json');
-        expect(response.payload).toMatchSnapshot();
+    it('returns 422 response if the URL is invalid', async () => {
+      const articleUrl = 'something-invalid';
+      files.getPageImages.mockImplementation(() => {
+        throw new Error('invalid-url');
```

Comment:
Awesome idea! The error dependencies around the app kinda gave me headaches already... 😉 

I think adding a file with the constants in `/services/util/` could also make sense, since the errors are used in various places (not necessarily **only** for file related things - `invalidUrl` is relevant for other endpoints as well).

What do you think (cc @juffel)?

Reply:
I really like the idea of putting them all into constants in some less specific directory like `/services/util`!